src/index.tsx

```diff
@@ -0,0 +1,242 @@
+
+    if (isInputValueValid(value)) {
+      changeCodeAtFocus(value);
+      focusInput(activeInput + 1);
+    } else {
+      const { nativeEvent } = event;
+      // @ts-expect-error - This was added previosly to handle and edge case
+      // for dealing with keyCode "229 Unidentified" on Android. Check if this is
+      // still needed.
+      if (nativeEvent.data === null && nativeEvent.inputType === 'deleteContentBackward') {
+        event.preventDefault();
+        changeCodeAtFocus('');
+        focusInput(activeInput - 1);
+      }
+    }
+  };
+
+  const handleFocus = (event: React.FocusEvent<HTMLInputElement>) => (index: number) => {
+    setActiveInput(index);
+    event.target.select();
+  };
+
+  const handleBlur = () => {
+    setActiveInput(activeInput - 1);
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    const otp = getOTPValue();
+    if (event.code === 'Backspace') {
+      event.preventDefault();
+      changeCodeAtFocus('');
+      focusInput(activeInput - 1);
+    } else if (event.code === 'Delete') {
+      event.preventDefault();
+      changeCodeAtFocus('');
+    } else if (event.code === 'ArrowLeft') {
+      event.preventDefault();
+      focusInput(activeInput - 1);
+    } else if (event.code === 'ArrowRight') {
+      event.preventDefault();
+      focusInput(activeInput + 1);
+    }
+    // React does not trigger onChange when the same value is entered
+    // again. So we need to focus the next input manually in this case.
+    else if (event.key === otp[activeInput]) {
+      event.preventDefault();
+      focusInput(activeInput + 1);
+    } else if (
+      event.code === 'Spacebar' ||
+      event.code === 'Space' ||
+      event.code === 'ArrowUp' ||
+      event.code === 'ArrowDown'
+    ) {
+      event.preventDefault();
+    }
+  };
+
+  const focusInput = (index: number) => {
+    const activeInput = Math.max(Math.min(numInputs - 1, index), 0);
+
+    if (inputRefs.current[activeInput]) {
+      inputRefs.current[activeInput]?.focus();
+      inputRefs.current[activeInput]?.select();
+      setActiveInput(activeInput);
+    }
+  };
+
+  const changeCodeAtFocus = (value: string) => {
+    const otp = getOTPValue();
+    otp[activeInput] = value[0];
+    handleOTPChange(otp);
+  };
+
+  const handleOTPChange = (otp: Array<string>) => {
+    const otpValue = otp.join('');
+    onChange(otpValue);
+  };
+
+  const handlePaste = (event: React.ClipboardEvent<HTMLInputElement>) => {
+    event.preventDefault();
+
+    const otp = getOTPValue();
+    let nextActiveInput = activeInput;
+
+    // Get pastedData in an array of max size (num of inputs - current position)
+    const pastedData = event.clipboardData
+      .getData('text/plain')
+      .slice(0, numInputs - activeInput)
+      .split('');
+
+    // Prevent pasting if the clipboard data contains non-numeric values
+    if (inputType === 'number' && pastedData.some((value) => isNaN(Number(value)))) {
```

Comment:
@prateek3255 I guess we should also be checking for `tel` too here

Reply:
Yes thanks for pointing out